schematic/utils/google_api_utils.py

```diff
@@ -145,3 +154,37 @@ def export_manifest_csv(file_path: str, manifest: Union[pd.DataFrame, str]) -> N
         manifest.to_csv(file_path, index=False)
     else:
         export_manifest_drive_service(manifest, file_path, mime_type="text/csv")
+
+
+def raise_final_error(retry_state: Any) -> Any:
+    """After the final attempt, raise the error.
+
+    Args:
+        retry_state (Any): retry state object
+
+    Returns:
+        Any: result of the outcome
+    """
+    return retry_state.outcome.result()
+
+
+@retry(
+    stop=stop_after_attempt(5),
+    wait=wait_chain(
+        *[wait_fixed(1) for i in range(2)]
+        + [wait_fixed(2) for i in range(2)]
+        + [wait_fixed(5)]
+    ),
+    retry=retry_if_exception_type(HttpError),
+    retry_error_callback=raise_final_error,
```

Comment:
We would need to do some more research on this portion `retry_error_callback` - We probably want this to raise the exception if this failed 5 times, we for sure don't want this to return back bad data.

I copied this out of the other location where we are using this `@retry` function